### PR TITLE
feat(rest-api-client): add `space.updateThread()` method

### PR DESCRIPTION
**Pull request by tuanphamcybozu**
## Why
- Support `updateThread` method to be able to update a thread of a space.

## What
- Add new method `updateThread`
- Add unit test
- Add demo script

## How to test
- Check test spec

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/master/CONTRIBUTING.md)
- [x] Updated documentation if it is required.
- [X] Added tests if it is required.
- [x] Passed `pnpm lint` and `pnpm test` on the root directory.

### CODE REVIEW

File: examples/rest-api-client-demo/src/space.ts [hunk 1]
```diff
@@ -33,4 +33,16 @@ export class Space {
       console.log(error);
     }
   }
+
+  public async updateThread() {
+    const body = "<b>This is an updated space body</b>";
+    const name = "Updated Thread Name";
+    try {
+      console.log(
+        await this.client.space.updateThread({ id: SPACE_ID, body, name }),
```
Comment (hung-cybo) [inline examples/rest-api-client-demo/src/space.ts:42]: @tuanphamcybozu 

The `id` parameter is Thread ID, not Space ID.
Could you check again?

File: packages/rest-api-client/src/client/SpaceClient.ts [hunk 2]
```diff
@@ -22,4 +22,15 @@ export class SpaceClient extends BaseClient {
     });
     return this.client.put(path, params);
   }
+
+  public updateThread(params: {
+    id: SpaceID;
```
Comment (hung-cybo) [inline packages/rest-api-client/src/client/SpaceClient.ts:27]: @tuanphamcybozu 

The type of this param is `ThreadID`, not `SpaceID`.

File: packages/rest-api-client/src/client/__tests__/SpaceClient.test.ts [hunk 3]
```diff
@@ -72,6 +72,26 @@ describe("SpaceClient", () => {
       expect(mockClient.getLogs()[0].params).toEqual(params);
     });
   });
+
+  describe("updateThread", () => {
+    const params = {
+      id: SPACE_ID,
```
Comment (hung-cybo) [inline packages/rest-api-client/src/client/__tests__/SpaceClient.test.ts:78]: This is `THREAD_ID`, not `SPACE_ID`.

File: examples/rest-api-client-demo/src/space.ts [hunk 4]
```diff
@@ -33,4 +33,16 @@ export class Space {
       console.log(error);
     }
   }
+
+  public async updateThread() {
+    const body = "<b>This is an updated space body</b>";
```
Comment (hung-cybo) [inline examples/rest-api-client-demo/src/space.ts:38]: This is just an example of test data, but I think It should be related to our API.

My suggestion:
`<b>This is an updated thread body</b>`

File: packages/rest-api-client/src/client/__tests__/SpaceClient.test.ts [hunk 5]
```diff
@@ -72,6 +72,26 @@ describe("SpaceClient", () => {
...
+      name: "Updated Thread Name",
+      body: "<b>This is an updated space body</b>",
```
Comment (hung-cybo) [inline packages/rest-api-client/src/client/__tests__/SpaceClient.test.ts:80]:
```suggestion
      body: "<b>This is an updated thread body</b>",
```

**Review by hung-cybo — COMMENTED** [review]
I have some comments.
Please check it.
Thank you.

File: packages/rest-api-client/src/client/types/index.ts [hunk 6]
```diff
@@ -2,6 +2,7 @@ export type AppID = string | number;
 export type RecordID = string | number;
 export type Revision = string | number;
 export type SpaceID = string | number;
+export type ThreadId = string | number;
```
Comment (hung-cybo) [inline packages/rest-api-client/src/client/types/index.ts:5]: Please upper case `Id` to `ID` as others.
`ThreadId` to `ThreadID`

**Review by hung-cybo — APPROVED** [review]
LGTM

**Review by tasshi-me — APPROVED** [review]
LGTM!

**Pull request merged**